pkg/cloudprovider/providers/gce/gce_addresses.go

```diff
@@ -85,3 +85,23 @@ func (gce *GCECloud) GetRegionAddress(name, region string) (*compute.Address, er
 	v, err := gce.service.Addresses.Get(gce.projectID, region, name).Do()
 	return v, mc.Observe(err)
 }
+
+// GetRegionAddressByIP returns the regional address matching the given IP
+// address. Note that if such an address does not exist, the function simply
+// returns a nil object. It does not return http.StatusNotFound like other
```

Comment:
I think the danger of someone not reading this is too great. Recommend we just create one of the errors of the same type. There's a function somewhere that does it for the fakes.